templates/primary-secondary/primary-secondary.js

```diff
@@ -988,6 +985,17 @@ class TemplatePrimarySecondary extends FocusVisiblePolyfillMixin(RtlMixin(LitEle
 				}
 			}
 		}
+		if (!this._secondary) {
```

Comment:
This approach resolves the elements not existing during `connectCallback` - but does try to re-add the listeners on every update. This might not be an issue because [duplicate event listeners by reference won't stack up](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#multiple_identical_event_listeners) and there doesn't seem to be a way to check if they exist, but if we're concerned we could add a boolean that flips when they're actually added and on `disconnectedCallback`

Reply:
Seems legit. @dbatiste I pointed Scy in this direction, let us know if its smelly.

Reply:
Yeah this is definitely a little precarious.  If one of those resizer objects is ever modified, to use an anonymous function for one of its handlers, to move the `bind` (which creates a new function) out of the constructor, or if a new resizer object is created, then the function instances would be different and the browser would not discard them as described.  This code really relies on someone changing that code with seemingly safe changes (which is somewhat abstracted from here) to know this code depends on them not making those changes.

I am certain the original author had reasons for writing it this way, however it would be much preferred to wire/unwire the `window` handlers in `connectedCallback`/`disconnectedCallback`, the local handlers in `render` with the `@` directive, and any handlers for `this` in `firstUpdated`.  Unfortunately it looks like that may be a little more involved.

It looks like the original code was relying on rAF to delay wire-up.  I think this was kind of risky already.  Did you ever get to the root of why adding the Localizer mixin caused the event wire-up to break in the connectedCallback?  Perhaps we just need to wait a task using `setTimeout` instead of rAF.

Aside from that, I see what you are doing now as one option.  Another option is to just do the wire-up in `firstUpdated` and if the component ever does need to support being removed and then re-added to the DOM, we cross that bridge then.  Neither of those options are amazing.

Reply:
I couldn't definitively find what was causing the problem when I went through it, but there were a few tests (nothing reproducible) where it magically disappeared, which makes me think it might be a race condition. I didn't get too far into debugging, though, it's pretty involved with the number of things happening here. I don't know the impact the localization logic has on when the dom loads but it does some asynchronous things in its connectedCallback and doesn't explicitly wait for them. I'm hoping not to sink too much more time into it right now though.

If we're ok with it not supporting connecting/disconnecting for now (that doesn't seem like an intuitive thing to do with a template like this anyway?) then firstUpdated seems like the cleanest quick resolution for now. Otherwise I can throw in the boolean I mentioned to allow it to reconnect this way without duplicating handlers, which is a bit messier but doesn't compromise the functionality. Any preference @dbatiste, since future work is more likely to land with the platform team?

Reply:
Yeah that sounds fine.